chia/full_node/full_node_store.py

```diff
@@ -52,7 +59,10 @@ class FullNodeStore:
     seen_unfinished_blocks: Dict[bytes32, None]
 
     # Unfinished blocks, keyed from reward hash
-    unfinished_blocks: Dict[bytes32, Tuple[uint32, UnfinishedBlock, PreValidationResult]]
+    # There may be multiple different unfinished blocks with the same partial
+    # hash (reward chain block hash). They are stored under their partial hash
+    # though. The inner dictionary uses the foliage hash as the key
+    unfinished_blocks: Dict[bytes32, Dict[Optional[bytes32], UnfinishedBlockEntry]]
```

Comment:
Since none is a valid key, we can only have one such None block in this dict? I guess that means those unfinished_blocks are the same if they don't have `foliage_transaction_block_hash`

Reply:
that's right. `None` either means it's not a transaction block, or that it's using the current protocol